### PR TITLE
Add config binder generator to ASP.NET Core transport pack

### PR DESCRIPTION
**Pull request by layomia**
Related https://github.com/dotnet/runtime/pull/84379, https://github.com/dotnet/sdk/pull/31654.
FYI @eerhardt @captainsafia

### CODE REVIEW

**Review by tarekgh — APPROVED** [review]

**Review by ericstj — APPROVED** [review]
LGTM - babysit this one when it hits ASP.NET by letting @wtgodbe know to ping you if it needs help when it hits ASP.NET Core.  You might need to manually replicate the exclusion in ASP.NET.

**Comment by layomia** [comment]
Blocked on https://github.com/dotnet/sdk/pull/31654.

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**
